### ISSUE TITLE
getShellVariableSetters() breaks on string containing single-quotes

### ISSUE THREAD

**Issue by 3541** [haikuports/haikuporter#145]
Setting up haikuporter for the first time, and when I attempt to run it with `./haikuporter`, it does the following:
```
Warning: Found old repository format - repopulating the repository ...
Populating repository ... 
        2048-1.0.0bash: line 45: -DCMAKE_INSTALL_SYSCONFDIR=/packages/2048-1.0.0-/.settings: No such file or directory
bash: line 144: local: can only be used in a function
bash: line 145: local: can only be used in a function
bash: -c: line 162: syntax error near unexpected token `}'
bash: -c: line 162: `}'

        2048_x86-1.0.0bash: line 45: -DCMAKE_INSTALL_SYSCONFDIR=/packages/2048_x86-1.0.0-/.settings: No such file or directory
bash: line 144: local: can only be used in a function
bash: line 145: local: can only be used in a function
bash: -c: line 162: syntax error near unexpected token `}'
bash: -c: line 162: `}'
```
Repeated for what appears to be every package in the repository.

**Comment by waddlesplash** [comment]
Ah, the downsides of using bash as a parser... although I've never seen this one before, odd.

**Comment by waddlesplash** [comment]
Can you share as much info about your system as possible? bash version, filesystem, etc.

**Comment by 3541** [comment]
Alright. All the information I can think of that could be relevant:

*

**Comment by 3541** [comment]
(ignore the close/reopen I clicked the wrong button)
All the information I can think of that might be relevant:
* Running haiku x86_gcc2 on nightly build hrev51568.
* Bash version 4.3.42
* Filesystem is BeFS
My haikuports.conf is as follows:
```
TREE_PATH="/boot/home/haikuports"
PACKAGER="Alex O'Brien <mail@domain.com>"
TARGET_ARCHITECTURE="x86_gcc2"
SECONDARY_TARGET_ARCHITECTURES="x86"
```
Let me know if there's anything more helpful I can give.

**Comment by mmuman** [comment]
Can you try without the single quote in the name?
Bash can get very confused with that kind of things...

**Comment by 3541** [comment]
Removing the apostrophe worked. Oh well.

**Comment by mmuman** [comment]
Well it shouldn't fail, it means something isn't quoted properly somewhere.

**Comment by 3541** [comment]
Revisiting this issue, I believe I tracked down the source of the problem.

At https://github.com/haikuports/haikuporter/blob/318743532ecdc8699f1290c13ae394affa606b69/HaikuPorter/ShellScriptlets.py#L52

`getShellVariableSetters` converts the dictionary of variables into shell variable definitions, simply surrounding the value in single quotes. However, Bash straight-up forbids the presence of single quotes in a single-quoted string. As such, the apostrophe in my name closes the string. As a quick hack, I replaced the relevant line with

```python
    result = '\n'.join("%s='%s'" % (k, v.replace("'", "'\\''"))
```

This allowed me to build packages with the `PACKAGER` value as described above. Although this does seem to fix the issue, I'm not sure if this is the correct place to do it, nor whether there might be a more elegant solution.
